request-server.go

```diff
@@ -279,9 +284,13 @@ func cleanPacketPath(pkt *sshFxpRealpathPacket) responsePacket {
 
 // Makes sure we have a clean POSIX (/) absolute path to work with
 func cleanPath(p string) string {
+	return cleanPathWithBase(p, "/")
+}
+
+func cleanPathWithBase(p, base string) string {
```

Comment:
The ordering of these arguments seems odd. We call `path.Join(base, p)`, and base is a prefix, but in this argument list it comes second?